src/cursor.rs

```diff
@@ -574,8 +583,12 @@ impl SyntaxNode {
     }
 
     #[inline]
-    pub fn green(&self) -> GreenNode {
-        self.green_ref().to_owned()
+    pub fn green(&self) -> Cow<'_, GreenNodeData> {
+        let green_ref = self.green_ref();
```

Comment:
`SyntaxNode::green_ref` should be marked unsafe, otherwise this looks too suspicious. I think it can be done later.

Reply:
Probably! I kind-of threw the towel on managing "internal" unsafety boundary here, but marking that one `unsafe` surely seems good!